File: tests/fixtures/sample.py
"""A tiny sample module to test our AST chunker."""

import os
import json

MAX_RETRIES = 3

class Config:
    """Holds app configuration."""

    def __init__(self, path):
        self.path = path

    def load(self):
        """Load config from disk as JSON."""
        with open(self.path) as f:
            return json.load(f)

    def save(self, data):
        with open(self.path, "w") as f:
            json.dump(data, f)


def retry(fn, attempts=MAX_RETRIES):
    """Retry a function up to `attempts` times."""
    for i in range(attempts):
        try:
            return fn()
        except Exception:
            if i == attempts - 1:
                raise

def main():
    cfg = Config("config.json")
    print(cfg.load())

File: tests/fixtures/sample.js
// A tiny sample module to test our JS chunker.
import fs from "fs";

const MAX_RETRIES = 3;

class Config {
  constructor(path) {
    this.path = path;
  }

  /** Load config from disk as JSON. */
  load() {
    return JSON.parse(fs.readFileSync(this.path, "utf-8"));
  }

  save(data) {
    fs.writeFileSync(this.path, JSON.stringify(data));
  }
}

/**
 * Retry a function up to `attempts` times.
 */
function retry(fn, attempts = MAX_RETRIES) {
  for (let i = 0; i < attempts; i++) {
    try {
      return fn();
    } catch (e) {
      if (i === attempts - 1) throw e;
    }
  }
}

const main = () => {
  const cfg = new Config("config.json");
  console.log(cfg.load());
};

export function notImplementedYet() {
  throw new Error("not implemented");
}
